model/testing/src/icon4py/model/testing/pytest_config.py

```diff
@@ -87,9 +86,11 @@ def pytest_addoption(parser):
 
 
 def pytest_runtest_setup(item):
-    for _ in item.iter_markers(name="datatest"):
-        if not item.config.getoption("--datatest"):
-            pytest.skip("need '--datatest' option to run")
+    match_marker(
+        item.own_markers,
+        item.config.getoption("--backend"),
```

Comment:
here we are passing the backend as `str`, but we can fix it by
```suggestion
        model_backends.BACKENDS[item.config.getoption("--backend")],
```